SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -136,6 +145,110 @@ void FLocalDeploymentManager::RefreshServiceStatus()
+	
+	// Now create and set up our sockets
+	FSocket* ListenSocket = SocketSubsystem->CreateSocket(NAME_Stream, TEXT("Runtime Port Test"), false /* bForceUDP */);
+	if (ListenSocket != nullptr)
+	{
+		ListenSocket->SetReuseAddr();
+		ListenSocket->SetNonBlocking();
+		ListenSocket->SetRecvErr();
+	
+		// Bind to our listen port
+		if (ListenSocket->Bind(*ListenAddr))
+		{
+			bSuccess = ListenSocket->Listen(0 /* MaxBacklog*/ );
+			ListenSocket->Close();
+		}
+		else
+		{
+			UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed to bind listen socket to addr (%s) for Runtime Port Test"), *ListenAddr->ToString(true));
+		}
+	}
+	else
+	{
+		UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed to create listen socket for Runtime Port Test"));
+	}
+
+	// Either we couldn't create the socket or couldn't listen on it so the port is probably bound.
+	return !bSuccess;
+}
+
+bool FLocalDeploymentManager::PreStartCheck()
+{
+	// Check for the known runtime port (5301) which could be blocked.
+	if (CheckIfPortIsBound(RequiredRuntimePort))
+	{
+	
+		// If it exists offer the user the ability to kill it.
+		if (FMessageDialog::Open(EAppMsgType::YesNo, LOCTEXT("KillPortBlockingProcess", "A required port is blocked by another process (potentially by an old deployment). Would you like to kill this process?")) == EAppReturnType::Yes)
+		{
+			const FString NetStatCmd = FString::Printf(TEXT("netstat"));
+
+			// -a display active tcp/udp connections, -o include PID for each connection, -n don't resolve hostnames
+			FString Args = TEXT("-n -o -a");
+			FString StdOut;
+			int32 ExitCode;
+			FString StdErr;
+			bool bSuccess = FPlatformProcess::ExecProcess(*NetStatCmd, *Args, &ExitCode, &StdOut, &StdErr);
+
+			if (ExitCode == ExitCodeSuccess && bSuccess)
+			{
+				
+				// Find which line of the output contains the port we're looking for
+				int PortIndex = StdOut.Find(FString::Printf(TEXT(":%i"), RequiredRuntimePort));
+				
+				// Throw away all lines before
+				FString PortLine = StdOut.Mid(PortIndex);
+				int PortLineEnd = PortLine.Find("\r\n");
+				
+				// Throw away all lines after
+				PortLine = PortLine.Mid(0, PortLineEnd);
+
+				// Match the PID
+				const FRegexPattern PidMatch(" +[0-9]+$");
+				FRegexMatcher RegMatcher(PidMatch, PortLine);
+				if (RegMatcher.FindNext())
+				{
+					int32 PidIndex = RegMatcher.GetMatchBeginning();
+					if (PidIndex >= 0)
+					{
+						
+						// Throw away everything except the PID
+						FString Pid = PortLine.RightChop(PidIndex);
+						Pid = Pid.TrimStart();
+						const FString TaskKillCmd = TEXT("taskkill");
+						Args = FString::Printf(TEXT("/F /PID %s"), *Pid);
+						bSuccess = FPlatformProcess::ExecProcess(*TaskKillCmd, *Args, &ExitCode, &StdOut, &StdErr);
+						if (bSuccess && ExitCode == ExitCodeSuccess)
+						{
+							return true;
+						}
+						UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed to kill process blocking required port. Error: %s"), *StdErr);
+					}
+				}
```

Comment:
If any of these statements fail we should be printing a message about what happened for debug purposes